pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -331,6 +358,60 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 	return &s, nil
 }
 
+func (pl *PodTopologySpread) calPodNotExistWithVolume(ctx context.Context,
+	pod *v1.Pod) (map[string][]*framework.PodInfo, error) {
+	allPvcs, err := pl.pvcs.PersistentVolumeClaims(pod.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("listing PersistentVolumeClaims: %w", err)
+	}
+	podNotExist := map[string][]*framework.PodInfo{}
+	processPVC := func(i int) {
+		pvc := allPvcs[i]
+		var nodeName string
+		if _, ok := pvc.Annotations[volume.AnnSelectedNode]; !ok {
+			return
+		}
+		nodeName = pvc.Annotations[volume.AnnSelectedNode]
+		var statefulSetReference metav1.OwnerReference
+		var podReference metav1.OwnerReference
+		for _, reference := range pvc.OwnerReferences {
+			if reference.Kind == "StatefulSet" {
+				statefulSetReference = reference
+			}
+			if reference.Kind == "Pod" {
+				podReference = reference
+			}
+		}
+		node, err := pl.sharedLister.NodeInfos().Get(nodeName)
+		if err != nil {
+			return
+		}
+		for _, podItem := range node.Pods {
+			if podItem.Pod.Namespace == pod.Namespace && podItem.Pod.Name == podReference.Name {
+				return
+			}
+		}
+		sts, err := pl.statefulSets.StatefulSets(pod.Namespace).Get(statefulSetReference.Name)
+		if err != nil {
+			return
+		}
+		podItem := &v1.Pod{
+			Spec: sts.Spec.Template.Spec,
+		}
+		podInfo := &framework.PodInfo{
+			Pod: podItem,
+		}
+		podInfo.Update(podItem)
```

Comment:
Why should we update the podInfo with sts's pod spec here?🤔️ I'm a little confused